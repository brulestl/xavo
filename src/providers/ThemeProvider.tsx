import React, { createContext, useContext, useState, useEffect, ReactNode } from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';

export interface ThemeColors {
  platinum: string;
  alabaster: string;
  saffron: string;
  eerieBlack: string;
  jet: string;
}

export interface ThemeSemanticColors {
  primary: string;
  primaryDisabled: string;
  background: string;
  cardBackground: string;
  textPrimary: string;
  textSecondary: string;
  accent: string;
  surface: string;
  border: string;
  shadow: string;
}

export interface Theme {
  colors: ThemeColors;
  semanticColors: ThemeSemanticColors;
  
  // Backward compatibility properties
  cardBackground: string;
  screenBackground: string;
  textPrimary: string;
  textSecondary: string;
  accent: string;
  surface: string;
  border: string;
  shadow: string;
  
  // Utility functions for component styling
  getBackgroundColor: () => string;
  getCardBackgroundColor: () => string;
  getPrimaryTextColor: () => string;
  getSecondaryTextColor: () => string;
  getAccentColor: () => string;
  getPrimaryButtonStyle: () => {
    backgroundColor: string;
    opacity: number;
  };
  getPrimaryButtonDisabledStyle: () => {
    backgroundColor: string;
    opacity: number;
  };
}

const colors: ThemeColors = {
  platinum: '#CFDBD5',
  alabaster: '#E8EDDF',
  saffron: '#F5CB5C',
  eerieBlack: '#242423',
  jet: '#333533',
};

const createTheme = (isDark: boolean): Theme => {
  const semanticColors: ThemeSemanticColors = {
    primary: colors.saffron,
    primaryDisabled: colors.saffron + '66', // 40% opacity
    background: isDark ? colors.eerieBlack : colors.alabaster,
    cardBackground: colors.platinum,
    textPrimary: isDark ? colors.alabaster : colors.eerieBlack,
    textSecondary: isDark ? colors.platinum : colors.jet,
    accent: colors.saffron,
    surface: isDark ? colors.jet : '#FFFFFF',
    border: isDark ? colors.jet : colors.platinum,
    shadow: isDark ? colors.eerieBlack : colors.jet,
  };

  return {
    colors,
    semanticColors,
    
    // Backward compatibility properties (mapped to semantic colors)
    cardBackground: semanticColors.cardBackground,
    screenBackground: semanticColors.background,
    textPrimary: semanticColors.textPrimary,
    textSecondary: semanticColors.textSecondary,
    accent: semanticColors.accent,
    surface: semanticColors.surface,
    border: semanticColors.border,
    shadow: semanticColors.shadow,
    
    getBackgroundColor: () => semanticColors.background,
    getCardBackgroundColor: () => semanticColors.cardBackground,
    getPrimaryTextColor: () => semanticColors.textPrimary,
    getSecondaryTextColor: () => semanticColors.textSecondary,
    getAccentColor: () => semanticColors.accent,
    getPrimaryButtonStyle: () => ({
      backgroundColor: semanticColors.primary,
      opacity: 1,
    }),
    getPrimaryButtonDisabledStyle: () => ({
      backgroundColor: semanticColors.primary,
      opacity: 0.4,
    }),
  };
};

interface ThemeContextType {
  theme: Theme;
  isDark: boolean;
  toggleTheme: () => void;
  // Tailwind class helpers
  getBackgroundClass: () => string;
  getCardBackgroundClass: () => string;
  getTextPrimaryClass: () => string;
  getTextSecondaryClass: () => string;
  getPrimaryButtonClass: () => string;
}

const ThemeContext = createContext<ThemeContextType | undefined>(undefined);

export const useTheme = () => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
};

interface ThemeProviderProps {
  children: ReactNode;
}

const THEME_STORAGE_KEY = '@app_theme_mode';

export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
  const [isDark, setIsDark] = useState(false);

  useEffect(() => {
    loadTheme();
  }, []);

  const loadTheme = async () => {
    try {
      const savedTheme = await AsyncStorage.getItem(THEME_STORAGE_KEY);
      if (savedTheme !== null) {
        setIsDark(savedTheme === 'dark');
      }
    } catch (error) {
      console.error('Error loading theme:', error);
    }
  };

  const toggleTheme = async () => {
    try {
      const newTheme = !isDark;
      setIsDark(newTheme);
      await AsyncStorage.setItem(THEME_STORAGE_KEY, newTheme ? 'dark' : 'light');
    } catch (error) {
      console.error('Error saving theme:', error);
    }
  };

  const theme = createTheme(isDark);

  // Tailwind class helpers for NativeWind
  const getBackgroundClass = () => isDark ? 'bg-eerie-black' : 'bg-alabaster';
  const getCardBackgroundClass = () => 'bg-platinum';
  const getTextPrimaryClass = () => isDark ? 'text-alabaster' : 'text-eerie-black';
  const getTextSecondaryClass = () => isDark ? 'text-platinum' : 'text-jet';
  const getPrimaryButtonClass = () => 'bg-saffron rounded-button text-button font-bold';

  return (
    <ThemeContext.Provider 
      value={{ 
        theme, 
        isDark, 
        toggleTheme,
        getBackgroundClass,
        getCardBackgroundClass,
        getTextPrimaryClass,
        getTextSecondaryClass,
        getPrimaryButtonClass,
      }}
    >
      {children}
    </ThemeContext.Provider>
  );
}; 