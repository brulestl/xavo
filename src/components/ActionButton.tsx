import React from 'react';
import { TouchableOpacity, Text, StyleSheet, View } from 'react-native';
import { useTheme } from '../providers/ThemeProvider';
import { Ionicons } from '@expo/vector-icons';

interface ActionButtonProps {
  title: string;
  icon?: keyof typeof Ionicons.glyphMap;
  onPress: () => void;
  disabled?: boolean;
}

export const ActionButton: React.FC<ActionButtonProps> = ({
  title,
  icon = 'chatbubble-outline',
  onPress,
  disabled = false,
}) => {
  const { theme, isDark } = useTheme();

  return (
    <TouchableOpacity
      style={[
        styles.button,
        {
          backgroundColor: disabled 
            ? theme.border 
            : isDark 
              ? theme.colors.jet 
              : theme.colors.saffron,
          opacity: disabled ? 0.6 : 1,
        }
      ]}
      onPress={onPress}
      disabled={disabled}
      activeOpacity={0.8}
    >
      <View style={styles.iconContainer}>
        <Ionicons
          name={icon}
          size={24}
          color="#FFFFFF"
        />
      </View>
      <Text style={[styles.title, { color: '#FFFFFF' }]}>
        {title}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    flex: 1,
    aspectRatio: 1,
    margin: 8,
    borderRadius: 16,
    padding: 16,
    justifyContent: 'center',
    alignItems: 'center',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  iconContainer: {
    marginBottom: 8,
  },
  title: {
    fontSize: 14,
    fontWeight: '600',
    textAlign: 'center',
    lineHeight: 16,
  },
}); 