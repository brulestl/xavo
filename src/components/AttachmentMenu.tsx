import React, { useRef, useEffect } from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  Animated,
  Dimensions,
  Modal,
  TouchableWithoutFeedback,
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { useTheme } from '../providers/ThemeProvider';

interface AttachmentMenuProps {
  visible: boolean;
  onClose: () => void;
  onTakePhoto: () => void;
  onChoosePhoto: () => void;
  onChooseFile: () => void;
  anchorPosition?: { x: number; y: number };
}

export const AttachmentMenu: React.FC<AttachmentMenuProps> = ({
  visible,
  onClose,
  onTakePhoto,
  onChoosePhoto,
  onChooseFile,
  anchorPosition = { x: 0, y: 0 },
}) => {
  const { theme } = useTheme();
  const scaleAnim = useRef(new Animated.Value(0)).current;
  const opacityAnim = useRef(new Animated.Value(0)).current;

  useEffect(() => {
    if (visible) {
      Animated.parallel([
        Animated.spring(scaleAnim, {
          toValue: 1,
          useNativeDriver: true,
          tension: 100,
          friction: 8,
        }),
        Animated.timing(opacityAnim, {
          toValue: 1,
          duration: 200,
          useNativeDriver: true,
        }),
      ]).start();
    } else {
      Animated.parallel([
        Animated.timing(scaleAnim, {
          toValue: 0,
          duration: 150,
          useNativeDriver: true,
        }),
        Animated.timing(opacityAnim, {
          toValue: 0,
          duration: 150,
          useNativeDriver: true,
        }),
      ]).start();
    }
  }, [visible]);

  const handleMenuItemPress = (action: () => void) => {
    onClose();
    // Delay the action slightly to allow the menu to close
    setTimeout(action, 150);
  };

  if (!visible) return null;

  const menuItems = [
    {
      icon: 'camera-outline',
      label: 'Take Photo',
      action: () => handleMenuItemPress(onTakePhoto),
      color: '#4CAF50',
    },
    {
      icon: 'image-outline',
      label: 'Choose Photo',
      action: () => handleMenuItemPress(onChoosePhoto),
      color: '#2196F3',
    },
    {
      icon: 'document-outline',
      label: 'Attach File',
      action: () => handleMenuItemPress(onChooseFile),
      color: '#FF9800',
    },
  ];

  return (
    <Modal transparent visible={visible} animationType="none">
      <TouchableWithoutFeedback onPress={onClose}>
        <View style={styles.overlay}>
          <Animated.View
            style={[
              styles.menuContainer,
              {
                backgroundColor: theme.semanticColors.surface,
                borderColor: theme.semanticColors.border,
                transform: [{ scale: scaleAnim }],
                opacity: opacityAnim,
                bottom: 120, // Position above the composer
                right: 20,
              },
            ]}
          >
            <View style={styles.menuContent}>
              {menuItems.map((item, index) => (
                <TouchableOpacity
                  key={item.label}
                  style={[
                    styles.menuItem,
                    {
                      borderBottomColor: theme.semanticColors.border,
                      borderBottomWidth: index < menuItems.length - 1 ? 1 : 0,
                    },
                  ]}
                  onPress={item.action}
                  activeOpacity={0.7}
                >
                  <View
                    style={[
                      styles.iconContainer,
                      { backgroundColor: item.color },
                    ]}
                  >
                    <Ionicons name={item.icon as any} size={20} color="#FFFFFF" />
                  </View>
                  <Text
                    style={[
                      styles.menuItemText,
                      { color: theme.semanticColors.textPrimary },
                    ]}
                  >
                    {item.label}
                  </Text>
                </TouchableOpacity>
              ))}
            </View>

            {/* Arrow pointing down to the attach button */}
            <View
              style={[
                styles.arrow,
                { borderTopColor: theme.semanticColors.surface },
              ]}
            />
          </Animated.View>
        </View>
      </TouchableWithoutFeedback>
    </Modal>
  );
};

const styles = StyleSheet.create({
  overlay: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.1)',
  },
  menuContainer: {
    position: 'absolute',
    borderRadius: 12,
    borderWidth: 1,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.15,
    shadowRadius: 8,
    elevation: 8,
    overflow: 'visible',
  },
  menuContent: {
    borderRadius: 12,
    overflow: 'hidden',
  },
  menuItem: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 16,
    paddingVertical: 14,
    minWidth: 160,
  },
  iconContainer: {
    width: 32,
    height: 32,
    borderRadius: 16,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 12,
  },
  menuItemText: {
    fontSize: 16,
    fontWeight: '500',
    flex: 1,
  },
  arrow: {
    position: 'absolute',
    bottom: -8,
    right: 24,
    width: 0,
    height: 0,
    borderLeftWidth: 8,
    borderRightWidth: 8,
    borderTopWidth: 8,
    borderLeftColor: 'transparent',
    borderRightColor: 'transparent',
  },
}); 