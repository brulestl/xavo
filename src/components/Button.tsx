import React from 'react';
import { TouchableOpacity, Text, StyleSheet, ViewStyle, TextStyle, TouchableOpacityProps } from 'react-native';
import { useTheme } from '../providers/ThemeProvider';

interface ButtonProps extends TouchableOpacityProps {
  title: string;
  variant?: 'primary' | 'secondary' | 'outline';
  size?: 'small' | 'medium' | 'large';
  fullWidth?: boolean;
}

export const Button: React.FC<ButtonProps> = ({
  title,
  variant = 'primary',
  size = 'medium',
  fullWidth = false,
  disabled = false,
  style,
  ...props
}) => {
  const { theme } = useTheme();

  const getButtonStyle = (): ViewStyle => {
    const baseStyle: ViewStyle = {
      borderRadius: 9999, // rounded-full equivalent
      alignItems: 'center',
      justifyContent: 'center',
      flexDirection: 'row',
    };

    // Size variations
    const sizeStyles = {
      small: { paddingVertical: 8, paddingHorizontal: 16 },
      medium: { paddingVertical: 12, paddingHorizontal: 24 },
      large: { paddingVertical: 16, paddingHorizontal: 32 },
    };

    // Variant styles
    const variantStyles = {
      primary: {
        backgroundColor: disabled ? `${theme.accent}66` : theme.accent,
      },
      secondary: {
        backgroundColor: theme.surface,
        borderWidth: 1,
        borderColor: theme.border,
      },
      outline: {
        backgroundColor: 'transparent',
        borderWidth: 1,
        borderColor: theme.accent,
      },
    };

    return {
      ...baseStyle,
      ...sizeStyles[size],
      ...variantStyles[variant],
      ...(fullWidth && { width: '100%' }),
    };
  };

  const getTextStyle = (): TextStyle => {
    const baseStyle: TextStyle = {
      fontWeight: '600',
      fontSize: 16,
    };

    const variantTextStyles = {
      primary: {
        color: theme.colors.eerieBlack,
      },
      secondary: {
        color: theme.textPrimary,
      },
      outline: {
        color: theme.accent,
      },
    };

    return {
      ...baseStyle,
      ...variantTextStyles[variant],
      ...(disabled && { opacity: 0.6 }),
    };
  };

  return (
    <TouchableOpacity
      style={[getButtonStyle(), style]}
      disabled={disabled}
      {...props}
    >
      <Text style={getTextStyle()}>{title}</Text>
    </TouchableOpacity>
  );
}; 