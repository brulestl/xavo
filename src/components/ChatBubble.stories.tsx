import type { Meta, StoryObj } from '@storybook/react';
import React from 'react';
import { View } from 'react-native';
import { ChatBubble } from './ChatBubble';
import { ThemeProvider } from '../providers/ThemeProvider';

const meta: Meta<typeof ChatBubble> = {
  title: 'Components/ChatBubble',
  component: ChatBubble,
  parameters: {
    layout: 'centered',
  },
  argTypes: {
    message: {
      control: 'text',
      description: 'The message content to display',
    },
    isUser: {
      control: 'boolean',
      description: 'Whether this is a user message or assistant message',
    },
    messageId: {
      control: 'text',
      description: 'Unique identifier for the message',
    },
    conversationId: {
      control: 'text',
      description: 'ID of the conversation this message belongs to',
    },
    timestamp: {
      control: 'text',
      description: 'Optional timestamp to display',
    },
    isStreaming: {
      control: 'boolean',
      description: 'Whether the message is currently being streamed',
    },
  },
  decorators: [
    (Story, context) => (
      <ThemeProvider>
        <View style={{ padding: 20, width: 300 }}>
          <Story {...context} />
        </View>
      </ThemeProvider>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof meta>;

const mockEditHandler = async (messageId: string, newContent: string) => {
  console.log(`Editing message ${messageId}: ${newContent}`);
  // Simulate API call delay
  await new Promise(resolve => setTimeout(resolve, 1000));
  return true;
};

export const UserMessage: Story = {
  args: {
    message: "How should I handle a difficult colleague who takes credit for my work?",
    isUser: true,
    messageId: "user-msg-1",
    conversationId: "conv-123",
    timestamp: "2:30 PM",
    onEditMessage: mockEditHandler,
    isStreaming: false,
  },
};

export const AssistantMessage: Story = {
  args: {
    message: "That's a challenging situation. Here are some strategies you can use to address this professionally while protecting your contributions...",
    isUser: false,
    messageId: "assistant-msg-1",
    conversationId: "conv-123",
    timestamp: "2:31 PM",
    isStreaming: false,
  },
};

export const StreamingMessage: Story = {
  args: {
    message: "I'm analyzing your situation and generating a response...",
    isUser: false,
    messageId: "assistant-msg-2",
    conversationId: "conv-123",
    timestamp: "2:32 PM",
    isStreaming: true,
  },
};

export const LongMessage: Story = {
  args: {
    message: "I've been working on a project for months, and my manager keeps giving all the credit to another team member during meetings. This person hasn't contributed much to the actual work, but they're very vocal about it in front of leadership. I'm frustrated and don't know how to address this without seeming petty or confrontational.",
    isUser: true,
    messageId: "user-msg-2",
    conversationId: "conv-123",
    timestamp: "2:30 PM",
    onEditMessage: mockEditHandler,
    isStreaming: false,
  },
};

export const WithoutEditCallback: Story = {
  args: {
    message: "This user message doesn't have edit capability enabled.",
    isUser: true,
    messageId: "user-msg-3",
    conversationId: "conv-123",
    timestamp: "2:33 PM",
    // No onEditMessage callback provided
    isStreaming: false,
  },
};