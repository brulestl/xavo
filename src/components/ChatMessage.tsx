import React from 'react';
import { Linking } from 'react-native';
import Markdown from 'react-native-markdown-display';
import { useTheme } from '../providers/ThemeProvider';
import { markdownStyles } from './markdownStyles';

interface ChatMessageProps {
  text: string;
}

export const ChatMessage: React.FC<ChatMessageProps> = ({ text }) => {
  const { theme } = useTheme();

  const handleLinkPress = (url: string) => {
    Linking.openURL(url).catch(err => {
      console.error('Failed to open URL:', err);
    });
  };

  return (
    <Markdown
      style={markdownStyles(theme.semanticColors)}
      mergeStyle={false}
      onLinkPress={handleLinkPress}
    >
      {text}
    </Markdown>
  );
}; 