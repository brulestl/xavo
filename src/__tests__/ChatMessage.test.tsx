import React from 'react';
import { render } from '@testing-library/react-native';
import { ChatMessage } from '../components/ChatMessage';
import { ThemeProvider } from '../providers/ThemeProvider';

// Mock the theme provider for testing
const MockThemeProvider = ({ children }: { children: React.ReactNode }) => (
  <ThemeProvider>
    {children}
  </ThemeProvider>
);

describe('ChatMessage', () => {
  it('renders markdown text correctly', () => {
    const sampleMarkdown = `
# This is a heading

This is **bold text** and this is *italic text*.

Here's a list:
- Item 1
- Item 2
- Item 3

And a numbered list:
1. First item
2. Second item
3. Third item

Here's some \`inline code\` and a [link](https://example.com).

> This is a blockquote

\`\`\`javascript
// This is a code block
function hello() {
  console.log("Hello world!");
}
\`\`\`
    `.trim();

    const { toJSON } = render(
      <MockThemeProvider>
        <ChatMessage text={sampleMarkdown} />
      </MockThemeProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders plain text correctly', () => {
    const plainText = 'This is just plain text without any markdown formatting.';

    const { toJSON } = render(
      <MockThemeProvider>
        <ChatMessage text={plainText} />
      </MockThemeProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('handles empty text', () => {
    const { toJSON } = render(
      <MockThemeProvider>
        <ChatMessage text="" />
      </MockThemeProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });
}); 