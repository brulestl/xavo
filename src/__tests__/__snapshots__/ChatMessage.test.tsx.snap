// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage handles empty text 1`] = `
<View
  style={
    {
      "margin": 0,
      "padding": 0,
    }
  }
/>
`;

exports[`ChatMessage renders markdown text correctly 1`] = `
<View
  style={
    {
      "margin": 0,
      "padding": 0,
    }
  }
>
  <View
    style={
      {
        "marginBottom": 8,
        "marginTop": 16,
      }
    }
  >
    <Text
      style={{}}
    >
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 20,
              "fontWeight": "700",
              "lineHeight": 22,
            },
            {},
          ]
        }
      >
        This is a heading
      </Text>
    </Text>
  </View>
  <View
    style={
      {
        "marginVertical": 4,
      }
    }
  >
    <Text
      style={{}}
    >
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {},
          ]
        }
      >
        This is 
      </Text>
      <Text
        style={
          {
            "color": "#1A1A1A",
            "fontWeight": "700",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#1A1A1A",
                "fontSize": 16,
                "fontWeight": "700",
                "lineHeight": 22,
              },
              {},
            ]
          }
        >
          bold text
        </Text>
      </Text>
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {},
          ]
        }
      >
         and this is 
      </Text>
      <Text
        style={
          {
            "color": "#1A1A1A",
            "fontStyle": "italic",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#1A1A1A",
                "fontSize": 16,
                "fontStyle": "italic",
                "lineHeight": 22,
              },
              {},
            ]
          }
        >
          italic text
        </Text>
      </Text>
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {},
          ]
        }
      >
        .
      </Text>
    </Text>
  </View>
  <View
    style={
      {
        "marginVertical": 4,
      }
    }
  >
    <Text
      style={{}}
    >
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {},
          ]
        }
      >
        Here’s a list:
      </Text>
    </Text>
  </View>
  <View
    style={
      {
        "marginLeft": 6,
        "marginVertical": 4,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "marginVertical": 2,
        }
      }
    >
      <Text
        accessible={false}
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {
              "color": "#1A1A1AAA",
              "marginRight": 6,
            },
          ]
        }
      >
        ·
      </Text>
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          style={{}}
        >
          <Text
            style={
              [
                {
                  "color": "#1A1A1A",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                {},
              ]
            }
          >
            Item 1
          </Text>
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginVertical": 2,
        }
      }
    >
      <Text
        accessible={false}
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {
              "color": "#1A1A1AAA",
              "marginRight": 6,
            },
          ]
        }
      >
        ·
      </Text>
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          style={{}}
        >
          <Text
            style={
              [
                {
                  "color": "#1A1A1A",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                {},
              ]
            }
          >
            Item 2
          </Text>
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginVertical": 2,
        }
      }
    >
      <Text
        accessible={false}
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {
              "color": "#1A1A1AAA",
              "marginRight": 6,
            },
          ]
        }
      >
        ·
      </Text>
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          style={{}}
        >
          <Text
            style={
              [
                {
                  "color": "#1A1A1A",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                {},
              ]
            }
          >
            Item 3
          </Text>
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "marginVertical": 4,
      }
    }
  >
    <Text
      style={{}}
    >
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {},
          ]
        }
      >
        And a numbered list:
      </Text>
    </Text>
  </View>
  <View
    style={
      {
        "marginLeft": 6,
        "marginVertical": 4,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "marginVertical": 2,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {
              "color": "#1A1A1AAA",
              "marginRight": 6,
            },
          ]
        }
      >
        1
        .
      </Text>
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          style={{}}
        >
          <Text
            style={
              [
                {
                  "color": "#1A1A1A",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                {},
              ]
            }
          >
            First item
          </Text>
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginVertical": 2,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {
              "color": "#1A1A1AAA",
              "marginRight": 6,
            },
          ]
        }
      >
        2
        .
      </Text>
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          style={{}}
        >
          <Text
            style={
              [
                {
                  "color": "#1A1A1A",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                {},
              ]
            }
          >
            Second item
          </Text>
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginVertical": 2,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {
              "color": "#1A1A1AAA",
              "marginRight": 6,
            },
          ]
        }
      >
        3
        .
      </Text>
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          style={{}}
        >
          <Text
            style={
              [
                {
                  "color": "#1A1A1A",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                {},
              ]
            }
          >
            Third item
          </Text>
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "marginVertical": 4,
      }
    }
  >
    <Text
      style={{}}
    >
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {},
          ]
        }
      >
        Here’s some 
      </Text>
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 4,
              "color": "#1A1A1A",
              "fontFamily": "monospace",
              "fontSize": 14,
              "paddingHorizontal": 4,
              "paddingVertical": 2,
            },
          ]
        }
      >
        inline code
      </Text>
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {},
          ]
        }
      >
         and a 
      </Text>
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#4285F4",
            "textDecorationLine": "underline",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#4285F4",
                "fontSize": 16,
                "lineHeight": 22,
                "textDecorationLine": "underline",
              },
              {},
            ]
          }
        >
          link
        </Text>
      </Text>
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {},
          ]
        }
      >
        .
      </Text>
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderLeftColor": "#4285F4",
        "borderLeftWidth": 4,
        "marginVertical": 8,
        "paddingHorizontal": 12,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "marginVertical": 4,
        }
      }
    >
      <Text
        style={{}}
      >
        <Text
          style={
            [
              {
                "color": "#1A1A1A",
                "fontSize": 16,
                "lineHeight": 22,
              },
              {},
            ]
          }
        >
          This is a blockquote
        </Text>
      </Text>
    </View>
  </View>
  <Text
    style={
      [
        {
          "color": "#1A1A1A",
          "fontSize": 16,
          "lineHeight": 22,
        },
        {
          "backgroundColor": "#f5f5f5",
          "borderColor": "#CCCCCC",
          "borderRadius": 4,
          "borderWidth": 1,
          "fontFamily": "Courier",
          "padding": 10,
        },
      ]
    }
  >
    // This is a code block
function hello() {
  console.log("Hello world!");
}
  </Text>
</View>
`;

exports[`ChatMessage renders plain text correctly 1`] = `
<View
  style={
    {
      "margin": 0,
      "padding": 0,
    }
  }
>
  <View
    style={
      {
        "marginVertical": 4,
      }
    }
  >
    <Text
      style={{}}
    >
      <Text
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 16,
              "lineHeight": 22,
            },
            {},
          ]
        }
      >
        This is just plain text without any markdown formatting.
      </Text>
    </Text>
  </View>
</View>
`;
